pages/words.js

```diff
@@ -0,0 +1,215 @@
+      <MenuItem key={key} value={value}>
+        {value}
+      </MenuItem>
+    )
+  )
+
+  const addPage = useCallback(async () => {
+    if (isLoading) return
+
+    setIsLoading(true)
+
+    const newWords = allWords.slice(
+      pageSize * pageIndex,
+      pageSize * (pageIndex + 1)
+    )
+    setWordsJsx((prevWords) => [...prevWords, wordsToJsx(newWords)])
+    setPageIndex((prevIndex) => prevIndex + 1)
+    setIsLoading(false)
+  }, [isLoading, allWords, pageIndex])
+
+  useEffect(() => {
+    setIsLoading(true)
+    setWordsJsx(wordsToJsx(allWords.slice(0, pageSize)))
+    setPageIndex(1)
+    setIsLoading(false)
+  }, [allWords])
+
+  useEffect(() => {
+    const loaderRefLocal = loaderRef.current
+    const observer = new IntersectionObserver((entries) => {
+      const target = entries[0]
+      if (target.isIntersecting) {
+        addPage()
+      }
+    })
+
+    if (loaderRefLocal) {
+      observer.observe(loaderRefLocal)
+    }
+
+    return () => {
+      if (loaderRefLocal) {
+        observer.unobserve(loaderRefLocal)
+      }
+    }
+  }, [addPage])
+
+  const onCharsChange = useCallback(
+    (value) => {
+      dispatch(setChars({ value }))
+    },
+    [dispatch]
+  )
+
+  const onSearchTypeChange = useCallback(
+    (value) => {
+      dispatch(setSearchType({ value }))
+    },
+    [dispatch]
+  )
+
+  const onLenIntervalChange = useCallback(
+    (value) => {
+      dispatch(setLenInterval({ value }))
+    },
+    [dispatch]
+  )
+
+  const onCaseInsensitiveChange = useCallback(
+    (value) => {
+      dispatch(setCaseInsensitive({ value }))
+    },
+    [dispatch]
+  )
+
+  const onDiacriticsInsensitiveChange = useCallback(
+    (value) => {
+      dispatch(setDiacriticsInsensitive({ value }))
+    },
+    [dispatch]
+  )
+
+  return (
+    <>
+      <Box className={layout_styles.page}>
+        <AppBar />
+        <Box
+          component="main"
+          className={layout_styles.main_decoder}
+          sx={{ color: 'primary.main' }}
+        >
+          <Box className={layout_styles.inputs_box}>
+            <Box>
+              <TextField
+                required
+                id="input-chars"
+                label="Hledané znaky"
+                value={chars}
+                onChange={(event) => onCharsChange(event.target.value)}
+                inputProps={{
+                  autocomplete: 'off',
+                  autocorrect: 'off',
+                  autocapitalize: 'off',
+                  spellcheck: 'false',
+                }}
+              />
+              <Select
+                value={searchType}
+                onChange={(event) => onSearchTypeChange(event.target.value)}
+              >
+                {filterMethodsJsx}
+              </Select>
+              <Slider
+                aria-label="Délka slova"
+                value={wordLenInterval}
+                onChange={(event, newValue) => onLenIntervalChange(newValue)}
+                min={1}
+                max={14}
```

Comment:
Trochu mě zarazila tahle arbitrární horní hranice. A věřím, že na většinu situací ti to bude stačit :-)

Reply:
Já váhám nad tím jak moc je tenhle slider vlastně užitečný. A asi u přesmyčky by se měl disableovat.